Source/DafnyLanguageServer/Workspace/ProjectManagerDatabase.cs

```diff
@@ -49,7 +49,10 @@ public class ProjectManagerDatabase : IProjectDatabase {
     }
 
     public async Task UpdateDocument(DidChangeTextDocumentParams documentChange) {
-      fileSystem.UpdateDocument(documentChange);
+      var success = fileSystem.UpdateDocument(documentChange);
+      if (!success) {
+        throw new ArgumentException("the document change ranges did not match ranges inside the documents");
```

Comment:
Nice.  Could you perhaps display somewhere the content of the document and the expected range? That might help or not live debugging. Or you are using it only for local debugging?

Reply:
I'll update it in a follow-up